src/lib.rs

```diff
@@ -8,55 +8,50 @@ extern crate strum;
 #[macro_use]
 extern crate strum_macros;
 
+pub mod config;
 pub mod policy_engine;
 pub mod types;
 pub mod verifier;
 
-use anyhow::{Context, Result};
-use policy_engine::opa;
-use serde_json::Value;
+use anyhow::{anyhow, Context, Result};
+use config::Config;
+use policy_engine::{PolicyEngine, PolicyEngineType};
 use std::collections::HashMap;
+use std::fs;
 use std::str::FromStr;
 use types::{AttestationResults, Evidence, TEE};
 
-#[macro_export]
-macro_rules! default_policy {
-    () => {
-        "../default_policy.rego"
-    };
-}
-
-#[derive(Debug, Default)]
+#[allow(dead_code)]
 pub struct AttestationService {
-    policy: String,
-    reference_data: String,
+    config: Config,
+    policy_engine: Box<dyn PolicyEngine + Send + Sync>,
+    reference_values: String,
 }
 
 impl AttestationService {
     /// Create a new Attestation Service instance.
     pub fn new() -> Result<Self> {
-        let policy = std::include_str!(default_policy!()).to_string();
+        let config = Config::default();
+        if !config.work_dir.as_path().exists() {
+            fs::create_dir_all(&config.work_dir)
+                .map_err(|e| anyhow!("Create AS work dir failed: {:?}", e))?;
+        }
+
+        let policy_engine = PolicyEngineType::from_str(&config.policy_engine)
+            .map_err(|_| anyhow!("Policy Engine {} is not supported", &config.policy_engine))?
+            .to_policy_engine(config.work_dir.as_path())?;
 
         let data: HashMap<String, Vec<String>> = HashMap::new();
-        //TODO: Reference data actually should be generated from RVPS storage when `evaluate` is called.
-        let reference_data = serde_json::json!({ "reference": data }).to_string();
+        //TODO: Reference values actually should be generated from RVPS storage when `evaluate` is called.
+        let reference_values = serde_json::json!({ "reference": data }).to_string();
 
         Ok(Self {
-            policy,
-            reference_data,
+            config,
+            policy_engine,
+            reference_values,
         })
     }
 
-    /// Get current policy.
-    pub fn policy(&self) -> String {
-        self.policy.clone()
-    }
-
-    /// Set customize policy.
-    pub fn set_policy(&mut self, policy: String) {
```

Comment:
So how do we set the policies now?

Reply:
Because the policy files of different policy engines have different formats, they are now set through the policy file path. For example, for `opa`, users can configure policy in path `/opt/confidential-containers/attestation-service/opa/policy.rego`, and the policy content in this file will be read during verification.

Reply:
Because the policy files of different policy engines have different formats, they are now set through the policy file path. For example, for `opa`, users can configure policy in path `/opt/confidential-containers/attestation-service/opa/policy.rego`, and the policy content in this file will be read during verification.